### PR TITLE
Cog 776 warn admin when permissions empty

### PR DESCRIPTION
**Pull request by Sushmareddy-OSI**
#### Which User story does this relate to (link)?
[https://osi-cwds.atlassian.net/browse/COG-776](https://osi-cwds.atlassian.net/browse/COG-776)

#### Brief feature description
- Warning with modal before leaving edit user screen to notify admin no permissions have been added
- renamed `modal => changeLogDetailsView` & 
    `pageHeaderButtons => DetailsPageHeaderButtons`

### CODE REVIEW

**Comment by Sushmareddy-OSI** [comment]
retest this please

File: app/javascript/selectors/detailSelector.js [hunk 1]
```diff
@@ -117,6 +117,11 @@ const checkForModifiedDetails = (initialDetails, updatedDetails) => {
   return initialDetails === updatedDetails ? undefined : updatedDetails
 }
 
+export const isPermissionsEmpty = state => {
+  const modifiedDetails = selectDetailRecords(state)
```
Comment (kelfink) [inline app/javascript/selectors/detailSelector.js]: naming this variable.... do we care / is it necessarily "modified" details?  Or is it just details ?

File: app/javascript/views/userDetail/NoPermissionsWarning.jsx [hunk 2]
```diff
@@ -0,0 +1,64 @@
+import React, { Component } from 'react'
+import PropTypes from 'prop-types'
+import { Modal, ModalBody, ModalFooter, ModalHeader, Button } from '@cwds/components'
+
+export default class ModalComponent extends Component {
```
Comment (kelfink) [inline app/javascript/views/userDetail/NoPermissionsWarning.jsx]: I think this class should be named after the file it's in.  It's not important that it's a ModalComponent (and we might be defining the same name other places, if it were). What we'll want to know down the road is that it's this NoPermissionsWarning object
```suggestion
export default class NoPermissionsWarning extends Component {
```

File: app/assets/stylesheets/application.scss [hunk 3]
```diff
@@ -122,6 +122,15 @@ input[type=number]::-webkit-outer-spin-button {
 
 /* Modal.jsx styles */
```
Comment (kelfink) [inline app/assets/stylesheets/application.scss]: are these common for multiple modal's ?

File: app/javascript/selectors/detailSelector.js (hunk 1 above)
Comment (Sushmareddy-OSI) [reply app/javascript/selectors/detailSelector.js]: it just details, updated!!

File: app/assets/stylesheets/application.scss (hunk 3 above)
Comment (Sushmareddy-OSI) [reply app/assets/stylesheets/application.scss]: No, it depends on design. The modal-body text font-size is different for changeLog Modal and NoPermissionsModal. So we have to add more class-names as per design requirement. Added that comment to know that these are related to **Modal**

Comment (kelfink) [reply app/assets/stylesheets/application.scss]: so I'm wondering why the class is .modal-body-text instead of specific, like nopermissionmodal-body-text

Comment (Sushmareddy-OSI) [reply app/assets/stylesheets/application.scss]: oh!! I didn't thought about the specific class-names.

Comment (Sushmareddy-OSI) [reply app/assets/stylesheets/application.scss]: updated!

Comment (kelfink) [reply app/assets/stylesheets/application.scss]: Yes!  much better thanks!

**Pull request merged**